tests/core/eth-module/test_transactions.py

```diff
@@ -12,20 +12,13 @@
     'make_chain_id, expect_success',
     (
         (
-            lambda web3: web3.version.network,
+            lambda web3: web3.net.chainId,
             True,
         ),
-        (
-            lambda web3: int(web3.version.network),
-            True,
-        ),
-        (
-            lambda web3: int(web3.version.network) + 1,
-            False,
-        ),
-        (
-            lambda web3: str(int(web3.version.network) + 1),
+        pytest.param(
+            lambda web3: 999999999999,
             False,
+            marks=pytest.mark.xfail,
```

Comment:
@carver why did you mark this as xfail?
```python
from web3.auto import w3
transaction = {'to': w3.eth.accounts[0], 'chainId': 9999999, 'from': w3.eth.accounts[1]}
w3.personal.unlockAccount(w3.eth.accounts[1], 'passphrase')
w3.eth.sendTransaction(transaction)
```
shouldn't the above code give `ValidationError` for `chainId`?  In fact validation middleware doesn't seem to do what it is expected to do.
I'm not very sure if this is a regression but I'm fixing this in #756.